platforms/android/run_module_app/build_apk.sh

```diff
@@ -0,0 +1,294 @@
+      fi
+      ;;
+    --inputs_file)
+      if [[ -n "$2" ]] && [[ ${2:0:1} != "-" ]]; then
+        IREE_INPUT_BUFFER_FILE=$(readlink -f $2)
+        shift 2
+      else
+        echo "Error: missing argument for $1" >&2
+        print_usage_and_exit
+      fi
+      ;;
+    -*|--*=) # Unsupported flags
+      echo "Error: Unsupported flag $1" >&2
+      exit 1
+      ;;
+    *) # Positional arguments
+      if [[ -z ${IREE_ARTIFACT_ROOT+x} ]]; then
+        IREE_ARTIFACT_ROOT=$(readlink -f $1)
+      else
+        echo "Error: <artifact-directory> already set to ${IREE_ARTIFACT_ROOT}" >&2
+        print_usage_and_exit
+      fi
+      shift
+      ;;
+  esac
+done
+
+if [[ -z ${IREE_ARTIFACT_ROOT+x} ]] || [[ -z ${IREE_INPUT_MODULE_FILE+x} ]] || \
+   [[ -z ${IREE_ENTRY_FUNCTION+x} ]] || [[ -z ${IREE_INPUT_BUFFER_FILE+x} ]] || \
+   [[ -z ${IREE_DRIVER+x} ]]; then
+  echo "Error: missing necessary parameters" >&2
+  print_usage_and_exit
+fi
+
+#################################
+# IREE Android app configurations
+#################################
+
+# The final Android APK name; default to "iree-run-module.apk".
+IREE_APK_NAME=${IREE_APK_NAME:-iree-run-module}
+# The CMAKE build type for compiling IREE. default to Release.
+IREE_BUILD_TYPE=${IREE_BUILD_TYPE:-Release}
+# The target Android API level; default to Android 10 (API level 29).
+IREE_ANDROID_API_LEVEL=${IREE_ANDROID_API_LEVEL:-29}
+# The target Android platform ABI; default to arm64-v8a.
+IREE_ANDROID_ABI=${IREE_ANDROID_ABI:-arm64-v8a}
+
+################################
+# Android SDK/NDK configurations
+################################
+
+# Android SDK root; must be provided as environment variable.
+# By default Android Studio uses
+# * $HOME/Android/Sdk for Linux.
+# * $HOME/Library/Android/sdk for macOS.
+# * $%LOCALAPPDATA%\Android\sdk for Windows.
+ANDROID_SDK_ROOT=${ANDROID_SDK_ROOT:?environment variable not set}
+# Android NDK root; must be provided as environment variable.
+ANDROID_NDK_ROOT=${ANDROID_NDK_ROOT:?environment variable not set}
+# Android SDK build tools version; default to empty given it's hard to
+# provide a default.
+# This typically should be provided by the caller and corresponds to a
+# directory under ${ANDROID_SDK_ROOT}/build-tools/, but there are cases where
+# an empty string is actually valid. For example, on Ubuntu, one can install
+# the SDK build tools via `apt install android-sdk-build-tools`, which will
+# then install it under /usr/lib/android-sdk/build-tools/.
+ANDROID_SDK_BUILD_TOOLS_VERSION=${ANDROID_SDK_BUILD_TOOLS_VERSION:-}
+# Key store for signing apk files; default to Android debug keystore created
+# by Android Studio.
+ANDROID_KEYSTORE=${ANDROID_KEYSTORE:-${HOME}/.android/debug.keystore}
+
+######################
+# IREE build toolchain
+######################
+
+CMAKE_BIN=${CMAKE_BIN:-$(which cmake)}
+NINJA_BIN=${NINJA_BIN:-$(which ninja)}
+CLANG_BIN=${CLANG_BIN:-$(which clang)}
+CLANGXX_BIN=${CLANGXX_BIN:-$(which clang++)}
```

Comment:
Not great to assume so much about the user's build system (that they're using ninja and clang)

Reply:
It's a bit unfortunate for sure but on Linux we aren't gonna support GCC well at the moment anyway. I guess on Windows this might be a problem given we want MSVC. Changed the variable to be `CC_BIN` and `CXX_BIN` at least to be less confusing.